app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java

```diff
@@ -123,85 +111,22 @@ public void choose() {
     }
 
     private void chooseNames() {
-        names.clearAnimation();
         textToSpeechManager.stopSpeaking();
         if (listState.getNumNames() > 0) {
-            List<String> chosenNames = listState.chooseNamesLegacy(settings);
-            chosenNamesText = NameUtils.flattenListToString(chosenNames, settings);
-
-            names.setAlpha(0.0f);
-            names.setText(chosenNamesText);
-
-            if (((AudioManager) getSystemService(Context.AUDIO_SERVICE))
-                    .getStreamVolume(AudioManager.STREAM_MUSIC) == 0) {
-                playNamesAnimation(defaultAnimationLengthMs);
-            } else {
-                playSound();
-            }
+            List<NameDO> chosenNames = listState.chooseNames(settings);
+            chosenNamesText = NameUtils.convertNameListToString(chosenNames, settings);
+            presentationAdapter.setChosenNames(chosenNames, settings.getShowAsList());
+            presentationManager.presentChosenName();
         } else {
             UIUtils.showLongToast(R.string.no_names_left, this);
         }
     }
 
-    private void playSound() {
-        try {
-            AssetFileDescriptor fileDescriptor = getAssets().openFd(DRUMROLL_FILE_NAME);
-            player.reset();
-            player.setDataSource(
-                    fileDescriptor.getFileDescriptor(),
-                    fileDescriptor.getStartOffset(),
-                    fileDescriptor.getLength());
-            player.prepare();
-            player.start();
-        } catch (Exception ex) {
-            UIUtils.showLongToast(R.string.drumroll_error, this);
+    @Override
+    public void speakNames() {
+        if (settings.getAutomaticTts()) {
```

Comment:
Ahhh, the speaking check is here. Calling speakNames() to have to sometimes not speak names is confusing and wastes a method call. I think we should just pass the boolean into presentation manager.